### PR TITLE
Fix CPP checks for CI

### LINKED ISSUE #148: [BUG] IWYU checks broken/disabled in CI
**Describe the bug**
Include what you use checks are broken in our current CI setup. In our prior setup it was installed in the container, but it isn't currently installed in our current images.

### PR DESCRIPTION
**Pull request by dagardner-nv**
Fixes the determination of the changed files list.
Adds IWYU to CI.

Fixes #148

### CODE REVIEW

File: ci/scripts/jenkins/checks.sh [hunk 1]
```diff
@@ -18,17 +18,38 @@ set -e
 
 source ${WORKSPACE}/ci/scripts/jenkins/common.sh
 
-rm -rf ${SRF_ROOT}/.cache/ ${SRF_ROOT}/build/
+rm -rf ${SRF_ROOT}/.cache/ ${SRF_ROOT}/build/ ${WORKSPACE_TMP}/iwyu
```
Comment (mdemoret-nv) [inline ci/scripts/jenkins/checks.sh:21]: CHange `${WORKSPACE_TMP}/iwyu` to `${IWYU_DIR}`

File: ci/scripts/jenkins/checks.sh [hunk 2]
```diff
@@ -18,17 +18,38 @@ set -e
...
 conda deactivate
 conda activate srf
 
 show_conda_info
 
+gpuci_logger "Installing IWYU"
+export IWYU_DIR="${WORKSPACE_TMP}/iwyu"
+cd ${WORKSPACE_TMP}
+git clone https://github.com/include-what-you-use/include-what-you-use.git ${IWYU_DIR}
+cd ${IWYU_DIR}
```
Comment (mdemoret-nv) [inline ci/scripts/jenkins/checks.sh:40]: Use `pushd` and `popd`

File: ci/scripts/jenkins/checks.sh [hunk 3]
```diff
@@ -18,17 +18,38 @@ set -e
...
+
 conda deactivate
 conda activate srf
 
 show_conda_info
 
+gpuci_logger "Installing IWYU"
+export IWYU_DIR="${WORKSPACE_TMP}/iwyu"
+cd ${WORKSPACE_TMP}
+git clone https://github.com/include-what-you-use/include-what-you-use.git ${IWYU_DIR}
```
Comment (mdemoret-nv) [inline ci/scripts/jenkins/checks.sh:39]:
```suggestion
git clone https://github.com/include-what-you-use/include-what-you-use.git ${IWYU_DIR}
```

**Review by mdemoret-nv — APPROVED** [review]
Approve with changes

**Review by mdemoret-nv — APPROVED** [review]

**Comment by dagardner-nv** [comment]
rerun tests

**Comment by dagardner-nv** [comment]
rerun tests

**Comment by dagardner-nv** [comment]
rerun tests

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**